### PR TITLE
Lecture 8

### PR DESCRIPTION
**Pull request by eugenevinitsky**
PDF and tex file for lecture on Krylov subspaces

### CODE REVIEW

File: notes/lecture8.tex [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+\item $i \neq j: \ p_i\trans A p_j = 0$: "Conjugacy" 
+\end{enumerate}
+\end{lemma}
+
+\begin{lemma}
+Let $\|u\|_A = \sqrt{\trspvec{u} A u}$ and $<u,v>_A = \trspvec{u} A v$ and $e_t = x^* - x_t$. Then $e_t$ minimizes $\|x^* - x\|_A$ over all vectors $x \in K_{t-1}$.
+\end{lemma}
+
+\begin{proof}
+We know that $x_t \in K_{t-1}$. Let $x \in K_{t-1}$. Define $x = x_t + \delta$. Then $e = x^* - x = x_t + \delta$. Lets compute the error in the A norm.
```
Comment (mrtzh) [inline notes/lecture8.tex]: Check spelling.
Use math environment for symbols like `A`.

File: notes/lecture8.tex [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+%\end{center}
+
+\subsection{Conjugate gradient method}
+We want to solve $Ax = b, \ A \geq 0$. 
+\begin{align*}
+x_0 = 0: \text{ "solution"} \\
+r_0 = b: \text{ "residual"} \\
+p_0 = r_0: \text{ "search direction"} \\
+\end{align*}
+For t = 1,2, ....
```
Comment (mrtzh) [inline notes/lecture8.tex]: Needs to go in math environment.

File: notes/lecture8.tex [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+\begin{align*}
+\eta_t = \frac{\|r_t\|}{<p_{t-1}, Ap_{t-1}>}: \text{ "step size"} \\
+x_t = x_{t-1} + \eta_t p_{t-1} \\
```
Comment (mrtzh) [inline notes/lecture8.tex]: Use `&` to align equations properly.

File: notes/lecture8.tex [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+We want to find a polynomial such that $p(A)b \approx \alpha_1 u_1$. Ideally, we would have \( p(\lambda_1) = 1\) and \( p(\lambda_i) = 0 \ \text{for } i > 1\). One thing that'll do this is if we get $p(\lambda_1) = 1$ and $\underset{i\ > 1}{\text{max}}p(\lambda_i)$ as small as possible. This will give us a close approximation to the top eigenvalue. \\
+
+What's an easy polynomial that'll get us pretty close to this? $p(\lambda) = \frac{\lambda^t}{\lambda_1 ^t}$. From this we get \(p(\lambda_1) = 1\) and \(p(\lambda_2) = (\frac{\lambda_2}{\lambda_1})^t\). We want \(p(\lambda_2)\) to get small so we care about how close \(\lambda_2\) is to $\lambda_1$.\\ 
+$\lambda_1 = (1 + \epsilon) \lambda_2$ then you need \(p(\lambda_2) = \frac{1}{(1+\epsilon)^t}\) if you want \(p(\lambda_2)\) to get small. 
+\begin{remark}[$\angle$ notation]
+\(\tan \angle (a,b)\) is the tangent of the angle between a and b
+\end{remark}
+\begin{theorem}[]
+$\tan \angle(p(A)b, u_1) \leq \underset{j > 1}{max} \frac{|p(\lambda_j)|}{|p(\lambda_1)|} \tan \angle (b, u)$
```
Comment (mrtzh) [inline notes/lecture8.tex]: Use `\max_{...}`

File: notes/lecture8.tex [hunk 5]
```diff
@@ -0,0 +1,219 @@
+\documentclass[12pt]{article}
```
[the remark this answers is not in the document]
Comment (mrtzh) [reply notes/lecture8.tex]: Please see `instructions.pdf` to make sure that your notes are consistent with those instructions.

File: notes/lecture8.tex [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+Basic & Gradient descent & Power methods \\ 
+\hline
+Accelerated & Chebyshev iteration & Chebyshev iteration \\
+\hline
+Accelerated and step size free & Conjugate gradient & Lanczos \\
+\hline
+\end{tabular}
+\end{center}
+\begin{remark}[Chebyshev]
+The Chebyshev flow requires step sizes to be carefully chosen while the "accelerated and step size free" methods do not.
```
Comment (mrtzh) [inline notes/lecture8.tex]: "flow" should be iteration. Latex quotes are ` `` ` and `''`

File: notes/lecture8.tex [hunk 7]
```diff
@@ -0,0 +1,219 @@
+\documentclass[12pt]{article}
+\usepackage{macros}
+\usepackage{tikz}
+\usepackage{pgfplots}
+\newtheorem{Fact}{Fact}
```
Comment (mrtzh) [inline notes/lecture8.tex]: There is already a `fact` environment in the macros.

File: notes/lecture8.tex [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+\begin{proof}
+We know that $x_t \in K_{t-1}$. Let $x \in K_{t-1}$. Define $x = x_t + \delta$. Then $e = x^* - x = x_t + \delta$. Lets compute the error in the A norm.
+\begin{align*}
+\|x^* - x\|_A^2 = \|e_t + \delta\|^{\trans} A(e_t + \delta) \\
+e = x^* - x \\
+e = x^* - x = e_t + \delta \\ 
+\|x^* - x\|_A^2 = e_t^{\trans}Ae_t + \delta^{\trans}A\delta + 2e_t^{\trans} A \delta \\
+A\delta \in K_{t-1}
+\end{align*}
+Want to argue that the last term $2e_t^{\trans} A \delta = 0$ because $e_t$ is orthogonal to the Krylov subspace. By definition $e_t^{\trans}A = r_t$.
```
Comment (mrtzh) [inline notes/lecture8.tex]: This is not a sentence.

**Comment by eugenevinitsky** [comment]
Thanks for the edits!

**Comment by eugenevinitsky** [comment]
Edits addressed.

**Comment by mrtzh** [comment]
This version doesn't even compile for me. Please make sure your notes can be compiled with `./compile.sh 8`.

File: notes/macros.sty [hunk 9]
```diff
@@ -303,6 +318,5 @@
 
 \newcommand{\domain}{\Omega}
 \newcommand{\epi}{\mathrm{epi}}
-\DeclareMathOperator*{\argmin}{\arg\!\min}
```
Comment (mrtzh) [inline notes/macros.sty]: Please do not edit macros.sty. This is breaking other files and preventing them from compiling. This macro for instance is used in lecture2.tex and elsewhere.

File: notes/macros.sty [hunk 10]
```diff
@@ -116,8 +124,6 @@
 \newcommand{\tableref}[1]{\hyperref[tab:#1]{Table~\ref{tab:#1}}}
 \newcommand{\itemlabel}[1]{\label{item:#1}}
 \newcommand{\itemref}[1]{\hyperref[item:#1]{Item~(\ref{item:#1})}}
-\newcommand{\examplelabel}[1]{\label{example:#1}}
```
Comment (mrtzh) [inline notes/macros.sty]: Why are you removing macros?

**Comment by mrtzh** [comment]
PS. Please make sure to pull before you submit the merge request. There have been numerous changes to the master branch since you opened the pull request. Thanks!

**Comment by eugenevinitsky** [comment]
Results compiled and master branch pulled in. Macros have been removed and other comments hopefully addressed.

**Comment by mrtzh** [comment]
It still didn't compile. Other lectures were broken. I had to fix it :-(

It doesn't look like you even tried to compile this. You also did a search & replace on all `<` signs and turned them into `\langle` signs, which is clearly not right. Same issue with `>`. 

Almost none of the comments I made were adequately addressed.

**Pull request merged**